### PR TITLE
[MP-6778] Adding Datepicker component

### PR DESCRIPTION
**Pull request by sathishkumar-thangavel**
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
 
## What should the SDET know?

### CODE REVIEW

File: src/components/datepicker/datepicker.tsx [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+  @State() endDateFormatted: any;
+  @State() dateHovered: any;
+
+  @Prop() mode: string;
+  @Prop() minDate: any;
+  @Prop() maxDate: any;
+  @Prop() startDate: any;
+  @Prop() endDate: any;
+  @Prop() dateFormat: string;
+  @Prop() dateValue: any;
```
[the remark this answers is not in the document]
Comment (sathishkumar-thangavel) [reply src/components/datepicker/datepicker.tsx]: @asif Please check my comment and let me know your thoughts.

File: src/components/select/select.scss [hunk 2]
```diff
@@ -67,6 +67,7 @@ label {
     letter-spacing: 0;
     line-height: 1.4;
     background-color: var(--input-bg);
+    width: 98%;
```
Comment (asifahmedfw) [inline src/components/select/select.scss:70]: Why 98%?

Comment (sathishkumar-thangavel) [reply src/components/select/select.scss:70]: Select component's input field's click area goes beyond the component width. It causes design breakage

Comment (asifahmedfw) [reply src/components/select/select.scss:70]: Add `box-sizing:border-box` with `width:100%`. See if that works.

**Pull request merged**